packages/protocol-sdk/src/client-base.ts

```diff
@@ -0,0 +1,71 @@
+import { foundry, zora, zoraTestnet } from "viem/chains";
+import type { BackendChainNames as BackendChainNamesType } from "./premint-api-client";
+import {
+  Chain,
+  PublicClient,
+  createPublicClient,
+  http,
+  parseEther,
+} from "viem";
+
+export type NetworkConfig = {
+  chainId: number;
+  zoraPathChainName: string;
+  zoraBackendChainName: BackendChainNamesType;
+  isTestnet: boolean;
+};
+
+export const REWARD_PER_TOKEN = parseEther("0.000777");
+
+export const BackendChainNamesLookup = {
+  ZORA_MAINNET: "ZORA-MAINNET",
+  ZORA_GOERLI: "ZORA-GOERLI",
+} as const;
+
+export const networkConfigByChain: Record<number, NetworkConfig> = {
+  [zora.id]: {
+    chainId: zora.id,
+    isTestnet: false,
+    zoraPathChainName: "zora",
+    zoraBackendChainName: BackendChainNamesLookup.ZORA_MAINNET,
+  },
+  [zoraTestnet.id]: {
+    chainId: zora.id,
+    isTestnet: true,
+    zoraPathChainName: "zgor",
+    zoraBackendChainName: BackendChainNamesLookup.ZORA_GOERLI,
+  },
+  [foundry.id]: {
+    chainId: foundry.id,
+    isTestnet: true,
+    zoraPathChainName: "zgor",
+    zoraBackendChainName: BackendChainNamesLookup.ZORA_GOERLI,
+  },
+};
+
+export abstract class ClientBase {
```

Comment:
id avoid having this base class just for holding the chain and network config.  how about just having the subclasses contain these objects and have utility functions to extract whats needed from both objects?

inheritance can be kind of hard to follow, and id be concerned that more stuff gets thrown in here than is needed.

Reply:
fine resolving this in a future PR (to get this out)

Reply:
The main reason I have things setup this way is the ease to override a network client without adding a ton of parameters.